Algorithms/mentoring/src/main/java/co/com/globant/java/mentoring/BasicAlgorithmExercises1.java

```diff
@@ -1,130 +1,65 @@
-        printResponse(generateResponse(highestAge,lowestAge,avarageAge,personsUnder18,personsOver18,personsSameAge,sameAgeCounter));
-    }
+import java.util.*;
+import java.util.stream.Collectors;
 
+public class LambdaEx1 {
     /**
-     * Display response at console
-     * @param response
+     * Basic Algorithm Exercises 1:
+     * Description: Given the following array, solve the following process.
+     * Get the highest Age.                         D
+     * Get the lowest Age.                          D
+     * Get the Average Age.                         D
+     * Get the number of persons under 18.          D
+     * Get the number of persons over 18.           D
+     * Get the number of persons of the same age.
+     * Every process must print the expected result on the console.*
      */
-    public void printResponse(StringBuilder response){
-        System.out.println(response);
-    }
 
-    /**
-     * Generate a StringBuilder that has the output message that shows the
-     * variables listed at the firm.
-     * @param highestAge        Int
-     * @param lowestAge         Int
-     * @param avarageAge        Int
-     * @param personsUnder18    Int
-     * @param personsOver18     Int
-     * @param personsSameAge    Dictionary
-     * @param sameAgeCounter    Int
-     * @return                  StringBuilder Response
-     */
-    public StringBuilder generateResponse(int highestAge,int lowestAge,int avarageAge,int personsUnder18,int personsOver18,Dictionary personsSameAge,int sameAgeCounter){
-        StringBuilder response = new StringBuilder("-- Results");
-        response.append("\n* The highest age -> ");
-        response.append(highestAge);
-
-        response.append("\n* The lowest age -> ");
-        response.append(lowestAge);
-
-        response.append("\n* The Average age -> ");
-        response.append(avarageAge);
-
-        response.append("\n* Persons under 18 -> ");
-        response.append(personsUnder18);
+    final Integer [] AGE = {87,6,17,37,11,89,34,26,25,21,21,37,55,33,81,6};
+    final static int AGE_THRESHOLD = 18;
+    List<Integer> ageList;
 
-        response.append("\n* Persons over 18 -> ");
-        response.append(personsOver18);
-
-        response.append("\n* The same age list -> ");
-        response.append(personsSameAge);
-
-        response.append("\n* The same age Counter -> ");
-        response.append(sameAgeCounter);
-
-        return response;
+    public LambdaEx1(){
+        ageList = Arrays.asList(AGE);
+    }
 
+    public void highestAge(){
+        Integer highest = ageList.stream().max(Integer::compare).get();
+        System.out.println("Highest -> " + highest);
+    }
+    public void lowestAge(){
+        int response =  ageList.stream().min(Integer::compare).get();
+        System.out.println("Lowest -> " + response);
+    }
+    public void avgAge(){
+        int response =  ageList.stream().reduce(Integer::sum).map(x -> x/ageList.size()).get();
```

Comment:
Try to enhance this method, because is not the optimal approach.

Reply:
Done, I remove the map phase, I put the division operation at the end.